theano/tensor/basic.py

```diff
@@ -4372,7 +4372,7 @@ def infer_shape(self, node, ishapes):
                     if ele == -1:
                         requ[i] = missing
             elif crit == 1:  # we reshape to -1
-                requ = [mul(*ishapes[0])]
+                requ = [mul(*ishapes[0])] if ishapes[0] else [1]
```

Comment:
I was talking about changing mul() interface do not crash when it get no inputs.

Can you tell more on what in the Reshape.infer_shape is broken? If it is broken, maybe we should wait for it to be fixed before converting Flatten to Reshape.

I don't understand your questions. For scalar (ndim=0), We simply have an empty tuple.